edb/common/span.py

```diff
@@ -300,3 +300,36 @@ def generic_visit(self, node):
         if getattr(node, 'span', None) is None:
             raise RuntimeError('node {} has no span'.format(node))
         super().generic_visit(node)
+
+
+def find_by_source_position(
```

Comment:
Add a comment somewhere around here that explains what we are looking for and sketches how. (The first node to contain the span in a post-order traversal of the AST, right? So, leaves preferred over their parents)